src/H5TSint.c

```diff
@@ -151,8 +153,8 @@ H5TS_term_package(void)
     H5TS_mutex_destroy(&H5TS_api_info_p.api_mutex);
     H5TS_atomic_destroy_uint(&H5TS_api_info_p.attempt_lock_count);
 
-    /* Clean up per-thread library info */
-    H5TS__tinfo_term();
+    /* Release critical section / mutex for modifying the thread info globals */
+    H5TS_mutex_destroy(&H5TS_tinfo_mtx_s);
```

Comment:
Why did you move this up?  Seems like it would be better where it was.

Reply:
Understood, but the global global resources for the thread-local info are created in H5TS__tinfo_init and it's more logical to destroy all of them in H5TS__tinfo_term.

Reply:
Fair point. If more threadlocal setup is ever added it might make sense to breakup tinfo_init() into a thread-local/thread-global counterparts to match this idea, but right now it makes sense to keep the mutex destroy where it is.